spec/constant_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe "Constant" do
+  it "Add two complex tensors." do
+    graph = Tensorflow::Graph.new
+    input1 = graph.constant( "const1",[Complex(2,2), Complex(2,34)], :complex)
+    input2 = graph.constant( "const2",[Complex(2,2), Complex(-32,22)], :complex)
+    graph.define_op("Add",'output', [input1, input2],"",nil)
+    graph.graph_def_raw = Tensorflow::GraphDef.encode(graph.graph_def)
+    session = Tensorflow::Session.new
+    session.extend_graph(graph)
+    result = session.run(nil, ["output"], nil)
+    expect(result[0]).to all_be_close([(4.0+4.0i), (-30.0+56.0i)])
+  end
```

Comment:
@Arafatk, speaking of specs, I'd recommend to scan through http://betterspecs.org/ and incorporate approaches, described there.

Basically, it is worth trying to make specs atomic and semantic. Something like this could be done:
```ruby
# what are we testing ? result of addition.
# so, we define subject
subject { result }
# and test it:
it { is_expected.to all_be_close([(4.0+4.0i), (-30.0+56.0i)]) }

# now, writing from target back to source.
# we need result to be calculated somehow, righ?..
let(:result) { session.run }
# ...and so, we need session
let(:session) { Session.new }
# ...and graph, probably?..
let(:graph) { Graph.new }
# ..and inputs, that we are checking!
let(:input1) { graph.constant( "const1",[Complex(2,2), Complex(2,34)], :complex) }
let(:input2) { graph.constant( "const2",[Complex(2,2), Complex(-32,22)], :complex) }

# ...and we need to wind everything up:
before {
  graph.define_op("Add",'output', [input1, input2],"",nil)
  graph.graph_def_raw = Tensorflow::GraphDef.encode(graph.graph_def)
  session.extend_graph(graph)
}
```

...And now we can see that the only difference with next tests is `define_op`. Soooo....
```ruby

describe 'Constants' do
  let(:graph) { Graph.new }

  let(:session) { Session.new }
  let(:result) { session.run }
  before {
    define_op
    graph.graph_def_raw = Tensorflow::GraphDef.encode(graph.graph_def)
    session.extend_graph(graph)
  }

  subject { result }

  context 'add' do
    let(:input1) { graph.constant( "const1",[Complex(2,2), Complex(2,34)], :complex) }
    let(:input2) { graph.constant( "const2",[Complex(2,2), Complex(-32,22)], :complex) }
    let(:define_op) { graph.define_op("Add",'output', [input1, input2],"",nil) }

    it { is_expected.to all_be_close([(4.0+4.0i), (-30.0+56.0i)]) }
  end

  context 'sub' do
    # If we could use the same inputs for all tests, it would be even more DRY
    let(:input1) { graph.constant( "const1",[634, 33], :float64) }
    let(:input2) { graph.constant( "const2",[332, 34], :float64) }
    let(:define_op) { graph.define_op("Sub",'output', [input1, input2],"",nil) }

    it { is_expected.to all_be_close([302, -1]) }
  end

  # ...and so on...
```